mongomock/collection.py

```diff
@@ -1534,7 +1534,7 @@ def _extend_collection(out_collection, field, expression):
                     _id = stage['$group']['_id']
                     if _id:
                         key_getter = functools.partial(_parse_expression, _id)
-                        out_collection = sorted(out_collection, key=key_getter)
+                        out_collection = sorted(out_collection, key=lambda x: str(key_getter(x)))
```

Comment:
This change doesn't seem related to the PR's title. Why was this introduced?

Reply:
hmmm, that was quite an old pr I can't fully recall why i did this...looks like I was adding this to make the test pass(it was broken)